src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exeption.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final String TABLE_MANUFACTURERS = "manufacturers";
+    private static final String COLUMN_ID = "id";
+    private static final String COLUMN_NAME = "name";
+    private static final String COLUMN_COUNTRY = "country";
+    private static final String COLUMN_IS_DELETED = "is_deleted";
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertFormatRequest = "INSERT INTO " + TABLE_MANUFACTURERS + " ("
+                + columnList(COLUMN_NAME, COLUMN_COUNTRY) + ") value(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createManufacturerStatement
+                        = connection.prepareStatement(insertFormatRequest,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Can not manufacturer to DB. Manufacturer: " + manufacturer,
+                    throwables);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String request = "SELECT * FROM " + TABLE_MANUFACTURERS + " WHERE "
+                + COLUMN_IS_DELETED + " = false AND id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacture = connection.prepareStatement(request,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            getManufacture.setLong(1, id);
+            ResultSet resultSet = getManufacture.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getManufacturerFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String selectAllQuery = "SELECT * FROM " + TABLE_MANUFACTURERS + " WHERE "
+                + COLUMN_IS_DELETED + " = 1";
```

Comment:
```suggestion
                + COLUMN_IS_DELETED + " = false";
```

[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html): Let’s use TRUE/FALSE instead of 1/0 with is_deleted column in the sql queries.

Reply:
done